app/src/main/java/org/p2p/wallet/newsend/smartselection/SendButtonStateManager.kt

```diff
@@ -0,0 +1,118 @@
+package org.p2p.wallet.newsend.smartselection
+
+import androidx.annotation.ColorRes
+import androidx.annotation.StringRes
+import android.content.res.Resources
+import java.math.BigDecimal
+import org.p2p.core.common.TextContainer
+import org.p2p.core.utils.Constants.SOL_SYMBOL
+import org.p2p.wallet.R
+import org.p2p.wallet.newsend.model.FeePayerState
+import org.p2p.wallet.newsend.model.FeePayerState.CalculationSuccess
+import org.p2p.wallet.newsend.model.FeePayerState.Failure
+import org.p2p.wallet.newsend.model.FeePayerState.FreeTransaction
+import org.p2p.wallet.newsend.model.FeePayerState.Idle
+import org.p2p.wallet.newsend.model.FeePayerState.NoStrategiesFound
+import org.p2p.wallet.newsend.model.FeePayerState.ReduceAmount
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason.CalculationError
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason.ExceededFee
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason.InputExceeded
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason.InvalidAmountForRecipient
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason.InvalidAmountForSender
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason.LowMinBalanceIgnored
+
+class SendButtonStateManager(
```

Comment:
maybe StateValidator or StateMapper?

Reply:
Yeah, I don't like the current name as well. 
will try something more